pydoctor/model.py

```diff
@@ -500,6 +500,10 @@ def msg(self, section, msg, thresh=0, topthresh=100, nonl=False, wantsnl=True, o
                 return
             else:
                 self.once_msgs.add((section, msg))
+
+        if thresh < 0:
+            self.warnings.setdefault(section, []).append((section, msg))
```

Comment:
maybe this needs a comment... or a reference to the ticket in which we implement a reported.

Anything below 0 is generated a  runtime warning / report

```
$ pydoctor --html-writer=pydoctor.test.InMemoryWriter pydoctor/test/testpackages/report_trigger -q
/home/adi/chevah/pydoctor/pydoctor/test/testpackages/report_trigger/__init__.py:8: Unknown field "bad_field"
/home/adi/chevah/pydoctor/pydoctor/test/testpackages/report_trigger/report_module.py:9: invalid ref to "BadLink" not resolved
```

Reply:
Maybe something like this:
```py
    if thresh < 0:
        # Count messages with a negative threshold as warnings.
        self.warnings.setdefault(section, []).append((section, msg))
```
When we move to a logger/reporter, it makes sense to derive the log level from the threshold, since they serve similar purposes.

Instead of `setdefault` with an empty list, you can use `defaultdict(list)` as the collection type, then you can just write `warnings[section].append(...)`.